reports/3sf.tex

```diff
@@ -0,0 +1,77 @@
+%! TeX root = report.tex
+
+\section{Basic 3SF concepts}\label{sec:3sf}
+
+In this section, we summarize the basic concepts of the 3SF protocol that this project depends on.
+We refer the reader to the 3SF paper~\cite{d20243} for a more comprehensive explanation or any concepts that this project depends on which we might have missed listing here.
+
+\paragraph*{Validators.} Participants of the protocol are referred to as \emph{validators} with $n$ being their total number.
+Every validator is identified by a unique cryptographic identity and the public keys are common knowledge. 
+Each validator has a \emph{stake} but for the purpose of this project each validator's stake is set to 1.
+% If a validator $v_i$ misbehaves and a proof of this misbehavior is given, it loses a part of its stake ($v_i$ gets \emph{slashed}). 
+
+\paragraph*{Slots.} Time is divided into \emph{slots}. 
+In ideal condition a new block (see below) is expected to be proposed at the beginning of each slot.
+
+\paragraph{Blocks and Chains.} 
+A \emph{block} is a pair of elements, denoted as \( B = (b,p) \). Here, \( b \) represents the \emph{block body} -- essentially, the main content of the block which contains a batch of transactions grouped together.
+Each block body contains a reference pointing to its \emph{parent} block. 
+The second element of the pair, \( p \geq 0 \), indicates the \emph{slot} where the block \( B \) is proposed.
+By definition, if $B_p$ is the parent of $B$, then $B_p.p < B.p$.
+The \emph{genesis block} is the only block that does not have a parent {and has a negative slot}.
+Given the definition above, each different block $B$ implicitly identifies a different finite \emph{chain} of blocks starting from block $B$, down to the genesis block, by recursively moving from a block to its parent.
+Hence, there is no real distinction between a block and the chain that it identifies.
+So, by chain $\chain$, we mean the chain identified by the block $\chain$.
+We write $\chain_1 \preceq \chain_2$ to mean that $\chain_1$ is a non-strict prefix of $\chain_2$.
+We say that $\chain_1$ \emph{conflicts} with $\chain_2$ if and only if neither $\chain_1 \preceq \chain_2$ nor $\chain_2 \preceq \chain_1$ holds. 
+
+\paragraph{Checkpoints.}
+In the protocol described in~\cite{d20243}, a \emph{checkpoint} is a tuple $(\chain, c)$, where \(\chain \) is a chain and \( c \) is a slot signifying where \( \chain \) is proposed for justification (this concept is introduced and explained below).
+However, for efficiency reasons, in the specification targeted by this project, a \emph{valid} checkpoint $\C$ is a triple $(H, c, p)$ where $H$ is the hash of a chain $\chain$, $c$ is as per the definition above and $p =  \chain.p$.
```

Comment:
```suggestion
However, for efficiency reasons, in the specification targeted by this project, a \emph{valid} checkpoint $\C$ is a triple $(H, c, p)$ where $H$ is the hash of a chain $\chain$, $c$ is the proposal slot as per the definition above and $p =  \chain.p$.
```

Reply:
I have taken on your suggestion but changed the wording a bit.